src/sw.ts

```
@@ -55,3 +55,131 @@ const serwist = new Serwist({
+      try {
+        const cache = await caches.open('quran-audio');
+        const response = await fetch(url, { signal: controller.signal });
+
+        if (!response.ok) {
+          event.ports[0]?.postMessage({ success: false, url, error: 'Network response was not ok' });
+          abortControllers.delete(url);
+          return;
+        }
+
+        const contentLength = response.headers.get('content-length');
+        const total = contentLength ? parseInt(contentLength, 10) : 0;
```

ESLint

[error] Prefer `Number.parseInt` over `parseInt`.

(unicorn/prefer-number-properties)